go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -421,12 +421,28 @@ func (agent *ActionAgent) demoteMaster(ctx context.Context, revertPartialFailure
 		}
 	}()
 
-	// Return the current replication position.
-	pos, err := agent.MysqlDaemon.MasterPosition()
+	// if a tablet was accepting writes, it cannot be replicating from another server,
+	// so we return the master position.
+	// We might call DemoteMaster on replicas in a no-master situation, and use the result to choose a master.
+	// So for replicas, we return the relay log position instead.
+	if !wasReadOnly {
+		// Return the current master position.
+		pos, err := agent.MysqlDaemon.MasterPosition()
+		if err != nil {
+			return "", err
+		}
+		return mysql.EncodePosition(pos), nil
+	}
+	rs, err := agent.MysqlDaemon.SlaveStatus()
 	if err != nil {
-		return "", err
+		// fall back to masterPosition
```

Comment:
I'd feel better if we only fall back if the error is specifically `ErrNotSlave`. If it's something else, we don't really know whether it's safe to fall back.

Reply:
Done